src/CodeGen_Hexagon.cpp

```diff
@@ -475,6 +515,11 @@ void CodeGen_Hexagon::compile_func(const LoweredFunc &f,
     debug(2) << "Lowering after forwarding stores:\n"
              << body << "\n\n";
 
+    // Replace div with long div if lanes > 1
+    debug(1) << "Replacing div with long_div...\n";
+    body = long_div(body);
+    body = common_subexpression_elimination(body);
```

Comment:
I think this is redundant with the common_subexpression_elimination in the long division implementation.

Reply:
Yeah. It's just that in the signed int case:
```
                Expr q = cast(a.type(), ulong_div(a_unsigned, b_unsigned));
                Expr a_neg = a >> make_const(a.type(), (a.type().bits() - 1));
                Expr b_neg = b >> make_const(a.type(), (a.type().bits() - 1));
```
a_neg and b_neg here are the same expressions created by lower_int_uint_div. That is why I thought of doing another round of cse.